sys/net/gnrc/transport_layer/tcp/gnrc_tcp_fsm.c

```diff
@@ -33,6 +33,11 @@
 #include "debug.h"
 
 /**
+ * @brief Flag indicating if a thread blocked by the GNRC TCP Api must be notified.
+ */
+static bool _notify_user;
```

Comment:
The return value for the fsm function is either:

1) negative for errors
2) positive for bytes sent/received
3) zero for everything is okay. 

And this information should be sent to the caller. The information, that a thread should be notified, is only relevant inside the fsm function and should not be sent to the callee. I think the global variable solution leads to less confusion and is much cleaner ...